connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -207,3 +208,23 @@ message Sample {
     int64 seed = 1;
   }
 }
+
+// Relation of type [[Range]] that generates a sequence of integers.
+message Range {
+  // Optional. Default value = 0
+  int32 start = 1;
+  int32 end = 2;
+  // Optional. Default value = 1
+  Step step = 3;
+  // Optional. Default value is assigned by 1) SQL conf "spark.sql.leafNodeDefaultParallelism" if
+  // it is set, or 2) spark default parallelism.
+  NumPartitions num_partitions = 4;
```

Comment:
There are two dimensions of things in this area:
1. Required versus Optional.
A field is required, meaning it must be set. A field can be optional. Meaning it could be set or not.

2. Field has default value or not.
A field can have a default value if not set. 


2. is an addition for 1.  If there is a field which is not set, there could be a default value to be used.

There are special cases that the default value for proto, is the same as the default value that Spark uses. In that case we don't need to differentiate the optionally. Otherwise we need this way to differentiate `set versus not set`, to adopt default values of Spark (unless we don't care the default values in Spark).

Reply:
To really answer your question: if we plan to respect default values for Spark for those optionally fields whose default proto values are different from Spark default values, this is the only way to respect default values for Spark.

Reply:
So in fewer words :) when `num_partitions` is an integer the default value is `0` even if it's not and for scalar types we can't differentiate between present or not. Understanding if `0` is a valid or invalid value defeats the purpose.

Thanks for the additional color!